src/main/java/com/powsybl/casestoreserver/CaseController.java

```diff
@@ -0,0 +1,167 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.casestoreserver;
+
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.xml.NetworkXml;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.powsybl.casestoreserver.CaseConstants.*;
+
+/**
+ * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ */
+@RestController
+@RequestMapping(value = "/" + CaseConstants.CASE_API_VERSION + "/case-server")
+@Api(value = "Case server")
+@ComponentScan(basePackageClasses = {CaseService.class})
+public class CaseController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CaseController.class);
+
+    @Inject
+    private CaseService caseService;
+
+    @GetMapping(value = "/cases")
+    @ApiOperation(value = "Get all cases")
+    public ResponseEntity<Map<String, String>> getCaseList() {
+        LOGGER.debug("Cases request received");
+
+        Map<String, String> cases;
+        try {
+            cases = caseService.getCaseList();
+        } catch (CaseException e) {
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, STORAGE_DIR_NOT_CREATED);
+        }
+        if (cases == null) {
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, IOEXCEPTION_MESSAGE);
+        }
+        return ResponseEntity.ok().body(cases);
+    }
+
+    @GetMapping(value = "/cases/{caseName:.+}")
+    @ApiOperation(value = "Get a case")
+    public ResponseEntity<StreamingResponseBody> getCase(@PathVariable("caseName") String caseName,
+                                     @RequestParam(value = "xiidm", required = false, defaultValue = "true") boolean xiidmFormat) {
+        LOGGER.debug("getCase request received with parameter caseName = {}", caseName);
+        if (xiidmFormat) {
+            Path file;
+            try {
+                file = caseService.getCase(caseName);
+            } catch (CaseException e) {
+                throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, STORAGE_DIR_NOT_CREATED);
```

Comment:
Why do you still need all this exception handling code???? 80% of this class code is related to exception and real code is totally cluttered by useless exception rethrow. In the case of a storage not created an exception with the right message is already thrown in service class. Why don't you let it just propagate? Conversion to HTTP status is  already done to 500 automatically and  in case it has to be refined, it could be done in an external exception handler (see link in my previous review)